### DISCUSSION TITLE
Disable default formatters

### DISCUSSION THREAD

**Discussion by YousefED** [microsoft/monaco-editor#3768] [Q&A] [answered]
Similar to how https://github.com/microsoft/monaco-json/pull/10 enables consumers to enable / disable modes (formatters), it would be useful to have this option for typescript / javascript as well.

I think at the moment `registerOnTypeFormattingEditProvider` is broken for monaco typescript, as only the default formatter will be called

**Comment by simerlec** [comment]
We have the same issue for `html` - the default formatter is still being used despite the use of a custom formatter.
```
monaco.languages.registerDocumentFormattingEditProvider(
'html',
documentFormatProviderHTML);
```

I was able to get around it by simply deferring the registration using a timeout which works in our specific use case but is less than ideal 🤷‍♂️ 
```
setTimeout(() => {
  monaco.languages.registerDocumentFormattingEditProvider(
  'html',
    documentFormatProviderHTML,
  );
}, 3000);
```

**Comment by troy351** [comment, marked as answer]
```js
const htmlDefaults = monaco.languages.html.htmlDefaults
htmlDefaults.setModeConfiguration({
    ...htmlDefaults.modeConfiguration,
    documentFormattingEdits: false,
    documentRangeFormattingEdits: false,
  })
```
this code can disable the default html formatter

**Comment by simerlec** [comment]
thank you very much @troy351
